File: Algorithm/Python/965. Univalued Binary Tree.py
# -*- coding: utf-8 -*-
# Created on Thu Jan 03 2019 9:24:48
# Author: WuLC
# EMail: liangchaowu5@gmail.com

# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isUnivalTree(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if root.left:
            same_left = root.val == root.left.val and self.isUnivalTree(root.left)
            if not same_left:
                return False
        if root.right:
            same_right = root.val == root.right.val and self.isUnivalTree(root.right)
            if not same_right:
                return False
        return True

File: Algorithm/Go/965. Univalued Binary Tree.go
/*
 * Created on Thu Jan 03 2019 9:44:29
 * Author: WuLC
 * EMail: liangchaowu5@gmail.com
 */

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isUnivalTree(root *TreeNode) bool {
	if root.Left != nil {
		same_left := root.Val == root.Left.Val && isUnivalTree(root.Left)
		if !same_left {
			return false
		}
	}
	if root.Right != nil {
		same_right := root.Val == root.Right.Val && isUnivalTree(root.Right)
		if !same_right {
			return false
		}
	}
	return true
}